Add git commit hooks via overcommit

monfresh opened the pull request:
**Why**:
- To enforce our commit message style guide rules
- To run the Slim linter before commits are pushed since Code Climate
does not have a Slim linting engine yet.

monfresh commented:
To test this, pull down this branch, then run `bundle install`, then make a change to any file ending in `.slim`, `git add` it, then attempt to `git commit`. The precommit hooks should run and let you know if anything went wrong.

Also try typing in a git commit message with greater than 50 characters in the subject line and greater than 72 characters in the body. It should display a warning, and won't save the commit message until it conforms to the text width rules.

sbc100 commented on Gemfile:
```diff
@@ -84,9 +82,13 @@ group :test do
   gem 'webmock'
 end
+
+group :linters do
```
Why not :development?

monfresh replied:
I just thought it would be nice to have the organized separately, but I just realized that we'll need to add yet another group to ignore when deploying, so perhaps I should keep them in development.

sbc100 commented:
Shouldn't overcommit be added to the Gemfile?

Also, doesn't one need to run 'overcommit --install' locally after a checkout?   Perhaps add this to the CONTRIBUTING.md (and maybe to bin/setup)?

sbc100 commented:
This looks really awesome BTW, I wasn't aware of this project before!

monfresh commented:
I don't think `overcommit --install` needs to be run by everyone. I think it's just supposed to be run once to configure the app and to add the `.overcommit.yml`.

Did you have to install the gem to get it to run the hooks?

sbc100 commented:
I'm pretty sure that the hooks need to be installed in .git/hooks otherwise they won't run.

overcommit --install seems to be the way to get them there.

I can verify this locally if you like...

monfresh commented:
Ah! You are right. I was under the impression the hooks would be checked into the repo. I'll update `bin/setup` and add overcommit to the Gemfile.

monfresh commented:
@sbc100 Check out the latest changes. I added overcommit to the Gemfile, installed it via bin/setup, and added a note in CONTRIBUTING about having to run `overcommit --sign` the first time you use it.

jgrevich commented:
It may be worth adding a `.gitmessage` config to help with the commit msg formatting. This is what I've been using:

```
## 57/70-character subject ############################## ############

__Why__

*

__How__

*

# 80-character wrapped description. ############################################
```

monfresh commented:
@jgrevich I'm not sure how to have everyone use the same `.gitmessage` since it also requires updating your `.gitconfig` which, when checked into the repo, will only take effect if each developer adds this to their `.git/config` inside the repo:
```
[include]
    path = /path-to-the-repo/.gitconfig
```

If you know of an easier way to have everyone use the same .gitmessage, let me know. Otherwise, let's merge this in and explore gitmessage later.

monfresh commented:
What I can do in the meantime is add instructions for setting up a .gitmessage in our CONTRIBUTING doc.

jgrevich commented:
Good point, instructions in CONTRIBUTING seems like the best spot. I thought we could add this to the root of the repo and it'd override `~/.gitmessage`. Apparently that's not the case and I also forgot that it requires the mod to the `.gitconfig`

```
[commit]
  template = ~/.gitmessage
```

monfresh commented:
@sbc100 I think this is ready to go now. Can you merge please?

sbc100 commented on .overcommit.yml:
```diff
@@ -0,0 +1,49 @@
+# For a complete list of options that you can use to customize hooks, see:
+# https://github.com/brigade/overcommit#configuration
+#
+# Uncomment the following lines to make the configuration take effect.
```
nit: I general I prefer not to have have things checked in and commented out, including instructions like this and defaults like at the end of the file.  Similar to the way we don't have the default config/routes.rb comments/instructions checked int.

Otherwise LGTM.  I don't feel too strongly about this, feel free to merge either way.

Not a big

monfresh commented:
Good point. I removed those instructions and the uncommented stuff at the end.
